modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPMetaDataHandler.java

```diff
@@ -55,4 +66,38 @@
 
         return underlying;
     }
+
+
+    /**
+     * Construct metadata while creating a copy of MessageMetadata for AMQP; after an update of the compression
+     * state of a message. This will accept parameters for buffer, original metadata and new compression state of the
+     * message.
+     *
+     * @param buf                       buffer of the original metadata
+     * @param originalMetadata          source metadata that needs to be copied
+     * @param newCompressedMessageValue Value to indicate if the message is compressed or not
+     * @return copy of the metadata as a byte array
+     */
+    public static byte[] constructMetadata(ByteBuffer buf, StorableMessageMetaData originalMetadata,
+                                           boolean newCompressedMessageValue) {
+        ContentHeaderBody contentHeaderBody = ((MessageMetaData) originalMetadata).getContentHeaderBody();
+        int contentChunkCount = ((MessageMetaData) originalMetadata).getContentChunkCount();
+        long arrivalTime = ((MessageMetaData) originalMetadata).getArrivalTime();
+        long sessionID = ((MessageMetaData) originalMetadata).getPublisherSessionID();
+
+        // modify message metadata, to update if the message is compressed or not
+        MessagePublishInfo messagePublishInfo = new CustomMessagePublishInfo(originalMetadata);
+        MessageMetaData modifiedMetaData = new MessageMetaData(messagePublishInfo, contentHeaderBody, sessionID,
+                contentChunkCount, arrivalTime, newCompressedMessageValue);
+
+        final int bodySize = 1 + modifiedMetaData.getStorableSize();
+        byte[] underlying = new byte[bodySize];
+        underlying[0] = (byte) modifiedMetaData.getType().ordinal();
+        buf = java.nio.ByteBuffer.wrap(underlying);
+        buf.position(1);
+        buf = buf.slice();
```

Comment:
Any specific reason to slice the buffer ? It might be good to explain the steps in comments wdyt ?